src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.cs

```diff
@@ -0,0 +1,412 @@
+        {
+            Debug.Assert(obj is Interop.Ole32.IStream);
+            Debug.Assert(s_wrapperEntry != null);
+
+            // Always return the same table mappings.
+            count = 1;
+            return s_wrapperEntry;
+        }
+
+        protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+        {
+            Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+
+            Guid pictureIID = IID_IPicture;
+            int hr = Marshal.QueryInterface(externalComObject, ref pictureIID, out IntPtr comObject);
+            if (hr == S_OK)
+            {
+                return new PictureWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        internal static class IStreamVtbl
+        {
+            public static IntPtr Create(IntPtr fpQueryInteface, IntPtr fpAddRef, IntPtr fpRelease)
+            {
+                IntPtr* vtblRaw = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IStreamVtbl), IntPtr.Size * 14);
+                vtblRaw[0] = fpQueryInteface;
+                vtblRaw[1] = fpAddRef;
+                vtblRaw[2] = fpRelease;
+                vtblRaw[3] = (IntPtr)(delegate* unmanaged<IntPtr, byte*, uint, uint*, int>)&Read;
+                vtblRaw[4] = (IntPtr)(delegate* unmanaged<IntPtr, byte*, uint, uint*, int>)&Write;
+                vtblRaw[5] = (IntPtr)(delegate* unmanaged<IntPtr, long, SeekOrigin, ulong*, int>)&Seek;
+                vtblRaw[6] = (IntPtr)(delegate* unmanaged<IntPtr, ulong, int>)&SetSize;
+                vtblRaw[7] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, ulong, ulong*, ulong*, int>)&CopyTo;
+                vtblRaw[8] = (IntPtr)(delegate* unmanaged<IntPtr, uint, int>)&Commit;
+                vtblRaw[9] = (IntPtr)(delegate* unmanaged<IntPtr, int>)&Revert;
+                vtblRaw[10] = (IntPtr)(delegate* unmanaged<IntPtr, ulong, ulong, uint, int>)&LockRegion;
+                vtblRaw[11] = (IntPtr)(delegate* unmanaged<IntPtr, ulong, ulong, uint, int>)&UnlockRegion;
+                vtblRaw[12] = (IntPtr)(delegate* unmanaged<IntPtr, Interop.Ole32.STATSTG*, Interop.Ole32.STATFLAG, int>)&Stat;
+                vtblRaw[13] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr*, int>)&Clone;
+
+                return (IntPtr)vtblRaw;
+            }
+
+            [UnmanagedCallersOnly]
+            private static int Read(IntPtr thisPtr, byte* pv, uint cb, uint* pcbRead)
+            {
+                try
+                {
+                    Interop.Ole32.IStream inst = ComInterfaceDispatch.GetInstance<Interop.Ole32.IStream>((ComInterfaceDispatch*)thisPtr);
+                    inst.Read(pv, cb, pcbRead);
+                }
+                catch (Exception e)
```

Comment:
By convention `e` is generally used for `EventArgs`, and `ex` for exceptions
```suggestion
                catch (Exception ex)
```

Ditto everywhere else